cmake/Platform/Mac/Install_mac.cmake

```diff
@@ -70,31 +64,74 @@ function(ly_install_target_override)
             COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME}
     )
 
+    set(install_relative_binaries_path "${ly_platform_install_target_RUNTIME_DIR}/${ly_platform_install_target_RUNTIME_SUBDIR}")
+
     if (${is_bundle})
         set_property(TARGET ${ly_platform_install_target_TARGET} PROPERTY RESOURCE ${cached_resources_dir})
+        set(runtime_output_filename "$<TARGET_FILE_NAME:${ly_platform_install_target_TARGET}>.app")
+
+        # For bundles, add a setreg file that contains install relative path to the bundle.
+        # This will be used when loading dynamic libraries outside the bundle.
+        string(CONFIGURE "${installed_binaries_path_template}" configured_setreg_file)
+        file(GENERATE
+            OUTPUT ${CMAKE_BINARY_DIR}/runtime_install/$<CONFIG>/${ly_platform_install_target_TARGET}BundlePath.setreg
+            CONTENT "${configured_setreg_file}"
+        )
+    else()
+        set(runtime_output_filename "$<TARGET_FILE_NAME:${ly_platform_install_target_TARGET}>")
+    endif()
+    
+    get_target_property(target_type ${ly_platform_install_target_TARGET} TYPE)
+    if(target_type IN_LIST LY_TARGET_TYPES_WITH_RUNTIME_OUTPUTS)
+        get_target_property(entitlement_file ${ly_platform_install_target_TARGET} ENTITLEMENT_FILE_PATH)
+        if (NOT entitlement_file)
+            set(entitlement_file "none")
+        endif()
+        
+        ly_file_read(${LY_ROOT_FOLDER}/cmake/Platform/Mac/runtime_install_mac.cmake.in template_file)
+        string(CONFIGURE "${template_file}" configured_template_file @ONLY)
+        file(GENERATE
+            OUTPUT ${CMAKE_BINARY_DIR}/runtime_install/$<CONFIG>/${ly_platform_install_target_TARGET}.cmake
+            CONTENT "${configured_template_file}"
```

Comment:
same here, if we are using these files during install but dont need to be installed, the question is, are these files different for monolithic vs non-monolithic? if they are, we need to key out of `LY_BUILD_PERMUTATION` as well so different build permutations (monolithic vs non-monolithic) dont overwrite files of each other
If they are the same file, then is fine to overwrite

Reply:
The files only apply to runtime targets. So for monolithic, it would only apply to our bundles. I guess they're different because it contains a path to the binary and that's going to be different based on monolithic vs non-monolithic. I'll update the path to include LY_BUILD_PERMUTATION.
I used `CMAKE_BINARY_DIR` because I didn't know a better location. What's the correct cmake variable/path to use for the setreg files?

Reply:
Also, what's a better path for placing install related generated files during the cmake configure/generate process? Is there an "install equivalent" to ${CMAKE_BINARY_DIR}?

Reply:
Yeah `${CMAKE_INSTALL_PREFIX}` is the variable that can be used.
Actually the `ly_install_files` command takes care of this

Reply:
These files are used during the install. They're not meant to be part of the install. So it has to be in the build tree right?

Reply:
Posted the earlier comment without understanding the original comment properly. I replied to that comment to explain why it needs to be generated there.

Reply:
Yes, if they are used during the install but not part of the install layout (result) then they should go to `CMAKE_BINARY_DIR`
Seems the only change needed here is to make them use the `LY_BUILD_PERMUTATION` since they are different for monolithic vs non-monolithic.

Reply:
Fixed